### PR TITLE
Add ability to change alias folder in NWNX_Object_Export

### PR DESCRIPTION
**Pull request by stenly87**

### CODE REVIEW

File: Plugins/Object/NWScript/nwnx_object.nss [hunk 1]
```diff
@@ -206,7 +206,8 @@ void NWNX_Object_RemoveIconEffect(object obj, int nIcon);
 /// @brief Export an object to the UserDirectory/nwnx folder.
 /// @param sFileName The filename without extension, 16 or less characters.
 /// @param oObject The object to export. Valid object types: Creature, Item, Placeable, Waypoint, Door, Store, Trigger
-void NWNX_Object_Export(object oObject, string sFileName);
+/// @param sAlias The alias of the resource directory to add the .git file to. Default: UserDirectory/nwnx
+void NWNX_Object_Export(object oObject, string sFileName, string sAlias = "NWNX");
```
Comment (Mingun) [inline Plugins/Object/NWScript/nwnx_object.nss:210]: Linux has case-sensitive filesystems by default. Are you sure that the default directory will be `nwnx`, not `NWNX`?

File: Plugins/Object/NWScript/nwnx_object.nss [hunk 2]
```diff
@@ -641,10 +642,11 @@ void NWNX_Object_RemoveIconEffect(object obj, int nIcon)
     NWNX_CallFunction(NWNX_Object, sFunc);
 }
 
-void NWNX_Object_Export(object oObject, string sFileName)
+void NWNX_Object_Export(object oObject, string sFileName, string sAlias = "NWNX")
 {
     string sFunc = "Export";
-
+	
+	NWNX_PushArgumentString(sAlias);
```
Comment (Mingun) [inline Plugins/Object/NWScript/nwnx_object.nss:649]: Here is tab instead of spaces

File: Plugins/Object/Object.cpp [hunk 3]
```diff
@@ -526,13 +526,22 @@ NWNX_EXPORT ArgumentStack Export(ArgumentStack&& args)
         const auto fileName = args.extract<std::string>();
           ASSERT_OR_THROW(!fileName.empty());
           ASSERT_OR_THROW(fileName.size() <= 16);
+		auto alias = args.extract<std::string>();
```
Comment (Mingun) [inline Plugins/Object/Object.cpp:529]: Here tab too

File: Plugins/Object/NWScript/nwnx_object.nss (hunk 1 above)
Comment (stenly87) [reply Plugins/Object/NWScript/nwnx_object.nss:210]: yeap. all folders in nwn are always upcased

File: Plugins/Object/Object.cpp [hunk 4]
```diff
@@ -526,13 +526,22 @@ NWNX_EXPORT ArgumentStack Export(ArgumentStack&& args)
         const auto fileName = args.extract<std::string>();
           ASSERT_OR_THROW(!fileName.empty());
           ASSERT_OR_THROW(fileName.size() <= 16);
+        auto alias = args.extract<std::string>();
```
Comment (Daztek) [inline Plugins/Object/Object.cpp:529]: Needs to be the following so it doesn't break prior compiled scripts:

```c++
std::string alias;
try
{
    alias = args.extract<std::string>();
}
catch(const std::runtime_error& e)
{
    LOG_WARNING("NWNX_Object_Export() called from NWScript without sAlias parameter. Please update nwnx_object.nss");
    alias = "NWNX";
}
```

Comment (stenly87) [reply Plugins/Object/Object.cpp]: Thanks

**Comment by Daztek** [comment]
Thank you for your contribution!

**Pull request merged**